src/utils/GcTcpHub.cxx

```diff
@@ -72,4 +62,12 @@ GcTcpHub::GcTcpHub(CanHubFlow *can_hub, int port)
 GcTcpHub::~GcTcpHub()
 {
     tcpListener_.shutdown();
+    // There is probably a race condition here with regard to the GC Port being
+    // closed elsewhere. Since shutting down a hub is typically proceeding an
+    // attempt at a graceful shutdown anyways, we are letting it go.
+    for (auto it = clients_.begin(); it != clients_.end(); ++it)
```

Comment:
We need to take a swap of the vector before calling these ::close calls. Otherwise we would have to do this with the atomic held to prevent callbacks from modifying the vector while we are iterating over it.
We can do a swap with a local vector under the atomic.

Reply:
I have a different fix for it.